operator/cmd/mesh/upgrade.go

```diff
@@ -164,6 +166,23 @@ func upgrade(rootArgs *rootArgs, args *upgradeArgs, l clog.Logger) (err error) {
 		return fmt.Errorf("failed to read the current Istio version, error: %v", err)
 	}
 
+	// Upgrade won't work, if the previous version of Istio is installed with revision
+	// Ref: https://github.com/istio/istio/issues/28566
+	pods, err := kubeClient.PodsForSelector(istioNamespace, "app=istiod")
+	if err != nil {
+		return fmt.Errorf("failed to fetch istiod pod, error: %v", err)
+	}
+	for _, pod := range pods.Items {
+		revision := pod.ObjectMeta.GetLabels()[label.IstioRev]
+		// If --revision is not passed, istio.io/rev: default
+		if revision != "" && revision != "default" {
```

Comment:
There is a slight difference between "there is a revisioned istiod (and a non-revisioned istiod)" and "all istiod are revisioned". I think we only need to stop the latter.

But we still need to let force flag to skip the check.

Reply:
Right, and it only stops if the previous istio is installed with the `revision` flag. If the istio is installed without a revision flag (which means its default) then it proceeds with a successful upgrade.